### ISSUE TITLE
Unable to open file

### ISSUE THREAD

**Issue by ttumkaya** [bnprks/BPCells#26]
Hi Ben,

Thank you for making such an amazing tool accessible to the community.

I'm using BPCells `open_matrix_10x_hdf5()` function to read an .h5 file as part of the Seurat v5 workflow, and having an error (please see below). 

The .h5 file is located in someone else's directory `(/outs/filtered_feature_bc_matrix.h5)` and I'm getting access as part of a group. Here are the settings for the directory and file itself: `drwxr-s---`, `-rw-r-----`.

Interestingly, I can read the same .h5 file from the same directory with `Read10x_h5()` function in Seurat with no issues.

Also, when I copy the .h5 into my own directory, then it works without any errors with the `open_matrix_10x_hdf5()` function.

Seems like there's a certain permission setting for the directory and/or file for BPCells to process it, and somehow that is not an issue for the Seurat function. 

Do you have any suggestions to fix this problem without having to copy all the data into my own directory (it's a pretty large dataset)?

Thank you!
Tayfun

```
Error in dims_matrix_10x_hdf5_cpp(path, buffer_size) : HDF5-API Errors:
    error #000: H5F.c in H5Fopen(): line 602: unable to open file
        class: HDF5
        major: File accessibilty
        minor: Unable to open file

    error #001: H5Fint.c in H5F_open(): line 990: unable to open file: time = Fri May  5 14:17:36 2023
, name = '/outs/filtered_feature_        class: HDF5
        major: File accessibilty
        minor: Unable to open file

    error #002: H5FD.c in H5FD_open(): line 991: open failed
        class: HDF5
        major: Virtual File Layer
        minor: Unable to initialize object

    error #003: H5FDsec2.c in H5FD_sec2_open(): line 337: unable to open file: name = '/outs/filtered_feature_bc_matrix.h5', errno = 13,        class: HDF5
        major: File accessibilty
```

**Comment by bnprks** [comment]
Hi Tayfun,

Thanks for the clear bug report! I think I've made a fix for your use-case in commit effa081. Could you try reinstalling again and confirming that it fixes your issue?

**Comment by ttumkaya** [comment]
Works like magic, thanks for the prompt fix, Ben!

Best,
Tayfun
